src/main/java/org/commcare/suite/model/QueryPrompt.java

```diff
@@ -156,11 +168,43 @@ public XPathExpression getRequired() {
         return required;
     }
 
+    @Nullable
+    public QueryPromptValidation getValidation() {
+        return validation;
+    }
+
     /**
      * @return whether the prompt has associated choices to select from
      */
     public boolean isSelect() {
         return getItemsetBinding() != null;
     }
 
+    /**
+     * Evalualtes validation message against given eval context
+     *
+     * @param ec eval context to evaluate the validation message
+     * @return evaluated validation message or a default text if no validation message is defined
+     */
+    public String getValidationMessage(EvaluationContext ec) {
+        if (validation != null && validation.getMessage() != null) {
+            return validation.getMessage().evaluate(ec);
+        }
+
+        try {
+            return Localization.get("case.search.answer.invalid");
```

Comment:
how come we are providing a default here but not for the required condition message: https://github.com/dimagi/commcare-core/pull/1139/files#diff-53ece50c900e231fe4323b42c57451b8b09b66d48f43cdf213dc08587c0a5bbdR194

Reply:
Mostly because HQ already has a default implemented for required messages but agree that FP should supply a default message for required condition as well. I will add it soon.